### PR TITLE
Hide icon only on hover & clean up hide all button styling

### PR DESCRIPTION
**Pull request by itsjoshthedeveloper**
@mich-m-li

### CODE REVIEW

**Comment by itsjoshthedeveloper** [comment]
hide all button before:
hide-all-before

hide all button after:
hide-all-after

**Comment by itsjoshthedeveloper** [comment]
hide icon on hover has some lag... @maxyuan6717 know how to fix this?

hide-icon-lag

**Comment by maxyuan6717** [comment]
the worksheet stuff is often plagued by a bunch of rerenders so that might be what's making it lag. Hopefully i can actually get the site working again on my machine and take a look

**Comment by hsheth2** [comment]
Yah we need to look into worksheet perf in general

@maxyuan6717 lmk if I can help with anything

**Comment by maxyuan6717** [comment]
Aight lag should be gone. In general, you don't wanna pass props to components if it's not completely necessary to prevent rerenders, so I just moved the hover logic to css

File: frontend/src/components/WorksheetListItem.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ function WorksheetListItem({
           <span className={styles.course_title}>{course.title}</span>
         </Col>
         {/* Hide Button */}
-        <Col xs="auto" className={'pl-0 pr-2 my-auto hidden'}>
```
Comment (itsjoshthedeveloper) [inline frontend/src/components/WorksheetListItem.js]: 😂

Comment (maxyuan6717) [reply frontend/src/components/WorksheetListItem.js]: we don't talk about that

Comment (itsjoshthedeveloper) [reply frontend/src/components/WorksheetListItem.js]: lol i was laughing at the commit name, typescript/eslint/prettier be making errors out of everything

**Comment by itsjoshthedeveloper** [comment]
i made sure that the hide icon stays visible which the course is hidden and that the tooltip is placed at the bottom so it won't cover the course title

@hsheth2 do you know what's causing the php check to fail?

**Comment by hsheth2** [comment]
The php check seems to be failing because it's started using php version 8 - you can ignore it here and I'll fix it in another branch

**Review by maxyuan6717 — APPROVED** [review]
fire

**Pull request merged**